mlkem/polyvec.c

```diff
@@ -273,3 +275,49 @@ void polyvec_add(polyvec *r, const polyvec *a, const polyvec *b)
         poly_add(&r->vec[i], &a->vec[i], &b->vec[i]);
     }
 }
+
+/*************************************************
+* Name:        polyvec_getnoise_eta1
+*
+* Description: Sample a polynomial deterministically from a seed and a nonce,
+*              with output polynomial close to centered binomial distribution
+*              with parameter KYBER_ETA1
+*
+* Arguments:   - polyvec *r: pointer to output vector of polynomial
+*              - const uint8_t *seed: pointer to input seed
+*                                     (of length KYBER_SYMBYTES bytes)
+*              - uint8_t nonce: one-byte input nonce
+**************************************************/
+void polyvec_getnoise_eta1(polyvec *r, const uint8_t seed[KYBER_SYMBYTES], uint8_t nonce)
+{
+    uint8_t buf[KYBER_ETA1 * KYBER_N / 4];
+    unsigned int i = 0;
+    for (i = 0; i < KYBER_K; i++)
+    {
+        prf(buf, sizeof(buf), seed, nonce++);
+        poly_cbd_eta1(&r->vec[i], buf);
```

Comment:
When `KYBER_K = 4 == KECCAK_WAY`, we can vectorize the whole function. `prf` is convert to `prfx4` and unroll `poly_cbd_eta1_x4`. 

This seem inefficient when `KYBER_K < KECCAK_WAY`, so I decided to unroll `poly_getnoise_eta1` in `indcpa.c` completely like `avx2`.